app/models/miq_report/search.rb

```diff
@@ -92,7 +92,19 @@ def paged_view_search(options = {})
 
     search_options = options.merge(:class => db, :conditions => conditions, :include_for_find => includes)
     search_options.merge!(:limit => limit, :offset => offset, :order => order) if order
-    search_options[:extra_cols] = va_sql_cols if va_sql_cols.present?
+
+    extra_cols = if includes
+                   includes.keys.map do |rel|
+                     if (rel = db_class.reflect_on_association(rel))
+                       # adding primary_key is probably overkill.
```

Comment:
You should always have the primary key just in case an association is needed on the model down the line, or you are immediately converting the results to a lower level data model.  I think in the case of reports, we aren't doing that, so the use of relations might be necessary.

I would say remove this comment, just so there is no confusion.